### ISSUE TITLE
Add a benchmark that heavily uses type annotations

### ISSUE THREAD

**Issue by ilevkivskyi** [python/performance#35]
This is a feature request.

Would it make sense to add a benchmark for code that _heavily_ uses annotations, generics, and other ``typing`` features? I even have a suggestion for such benchmark: running ``mypy`` on itself (on its own code). The run currently takes around 10-20 sec. (depending on machine). If this is too long, other option would be to run mypy on one of typeshed stubs, this will take less than a second.

**Comment by vstinner** [comment]
I'm unable to write such benchmark. Do you want to write such benchmark?

**Comment by ilevkivskyi** [comment]
> Do you want to write such benchmark?

Yes, I think I can. Just wanted to check if there is interest in such benchmark, IIUC there is.
